cosmos/module_client.go

```diff
@@ -0,0 +1,286 @@
+	"github.com/mesg-foundation/engine/x/service"
+)
+
+// ModuleClient handles all communication with every module.
+type ModuleClient struct {
+	*Client
+}
+
+// NewModuleClient creates new module client.
+func NewModuleClient(c *Client) *ModuleClient {
+	return &ModuleClient{Client: c}
+}
+
+func sroutef(format string, args ...interface{}) string {
+	return fmt.Sprintf("custom/"+format, args...)
+}
+
+// CreateService creates a new service from definition.
+func (mc *ModuleClient) CreateService(req *api.CreateServiceRequest) (*servicepb.Service, error) {
+	acc, err := mc.GetAccount()
+	if err != nil {
+		return nil, err
+	}
+	msg := service.NewMsgCreateService(acc.GetAddress(), req)
+	tx, err := mc.BuildAndBroadcastMsg(msg)
+	if err != nil {
+		return nil, err
+	}
+	return mc.GetService(tx.Data)
+}
+
+// GetService returns the service that matches given hash.
+func (mc *ModuleClient) GetService(hash hash.Hash) (*servicepb.Service, error) {
+	var out *servicepb.Service
+	route := sroutef("%s/%s/%s", service.QuerierRoute, service.QueryGetService, hash)
+	return out, mc.QueryJSON(route, nil, &out)
+}
+
+// ListService returns all services.
+func (mc *ModuleClient) ListService() ([]*servicepb.Service, error) {
+	var out []*servicepb.Service
+	route := sroutef("%s/%s", service.QuerierRoute, service.QueryListService)
+	return out, mc.QueryJSON(route, nil, &out)
+}
+
+// ExistService returns if a service already exists.
+func (mc *ModuleClient) ExistService(hash hash.Hash) (bool, error) {
+	var out bool
+	route := sroutef("%s/%s/%s", service.QuerierRoute, service.QueryExistService, hash)
+	return out, mc.QueryJSON(route, nil, &out)
+}
+
+// HashService returns the calculate hash of a service.
+func (mc *ModuleClient) HashService(req *api.CreateServiceRequest) (hash.Hash, error) {
+	var out hash.Hash
+	route := sroutef("%s/%s", service.QuerierRoute, service.QueryHashService)
+	return out, mc.QueryJSON(route, req, &out)
+}
+
+// CreateProcess creates a new process.
+func (mc *ModuleClient) CreateProcess(req *api.CreateProcessRequest) (*processpb.Process, error) {
+	acc, err := mc.GetAccount()
+	if err != nil {
+		return nil, err
+	}
+	msg := process.NewMsgCreateProcess(acc.GetAddress(), req)
+	tx, err := mc.BuildAndBroadcastMsg(msg)
+	if err != nil {
+		return nil, err
+	}
+	return mc.GetProcess(tx.Data)
+}
+
+// GetInstance returns the instance that matches given hash.
+func (mc *ModuleClient) GetInstance(hash hash.Hash) (*instancepb.Instance, error) {
+	var out *instancepb.Instance
+	route := sroutef("%s/%s/%s", instance.QuerierRoute, instance.QueryGetInstance, hash)
+	return out, mc.QueryJSON(route, nil, &out)
+}
+
+// ListInstance returns all instances.
+func (mc *ModuleClient) ListInstance(req *api.ListInstanceRequest) ([]*instancepb.Instance, error) {
+	var out []*instancepb.Instance
+	route := sroutef("%s/%s", instance.QuerierRoute, instance.QueryListInstances)
+	return out, mc.QueryJSON(route, req, &out)
```

Comment:
Only the filter should be passed to the query

original: https://github.com/mesg-foundation/engine/pull/1650/files#diff-cfb2b44e2ea93e1c0e4199d390e79f83L31
```diff
-	if err := s.client.QueryJSON(fmt.Sprintf("custom/%s/%s", instance.QuerierRoute, instance.QueryListInstances), request.Filter, &instances); err != nil {	
```

Reply:
Strange it should panic, but I've fixed it.